shared/analytics/README.md

```diff
@@ -0,0 +1,66 @@
+### Event types follow the Object Action pattern
+E.g. `Community Created`
+
+### Modifiers can be added to event types to clarify meaning
+E.g. `Message Sent Failed`
+
+### Events should contain as much metadata as possible for more comprehensive downstream analysis. 
+If an event can be evaluated in the context of a parent entity (community, channel, or thread), that parent data should be included. Include timestamps, actors, entities, and contextually-revelant strings (e.g. when tracking search results with 0 results, be sure to track the query string itself).
```

Comment:
I know this might sound like a good idea, but we should do anonymous analytics—let's avoid sending an personally identifiable information at all costs.

Because first, GDPR would require us to get explicit consent to this tracking (i.e. we have to present a checkbox and users have to opt-in!), and second it's just the right thing to do. We don't gain that much from that data. 

----

If there is a lot for us to gain from knowing how many users did something, we could work around this by hashed the `userId` before sending it. (probably bcrypt) The same `userId` will get the same hashed output every time, so it'll be consistent at the provider's data. Yet, neither the provider nor we can look at a specific event and figure out which specific user in our database did it, since hashes are one-way!